test/source_test.go

```diff
@@ -65,3 +67,114 @@ func testCmdSource(t *testing.T) {
 		return gwclient.NewResult(), nil
 	})
 }
+
+func testSourceMountPathHandledBySource(t *testing.T) {
+	t.Parallel()
+	ctx := startTestSpan(t)
+
+	sourceName := "checkcmd"
+	spec := &dalec.Spec{
+		Name: "cmd-source-ref",
+		Sources: map[string]dalec.Source{
+			sourceName: {
+				Path: "/baz",
+				DockerImage: &dalec.SourceDockerImage{
+					Ref: "busybox:latest",
+					Cmd: &dalec.Command{
+						Mounts: []dalec.SourceMount{
+							{
+								Dest: "/dst",
+								Spec: dalec.Source{
+									Path: "/out",
+									DockerImage: &dalec.SourceDockerImage{
+										Ref: "busybox:latest",
+										Cmd: &dalec.Command{
+											Steps: []*dalec.BuildStep{
+												{
+													Command: "mkdir -p /out; echo 'test contents' > /out/file.txt",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						Steps: []*dalec.BuildStep{
+							{
+								Command: `set -ex`,
+							},
+							{
+								Command: "mkdir -p /baz",
+							},
+							{
+								Command: `cp /dst/file.txt /baz`,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testEnv.RunTest(ctx, t, func(ctx context.Context, gwc gwclient.Client) (*gwclient.Result, error) {
+		req := newSolveRequest(withBuildTarget("debug/sources"), withSpec(ctx, t, spec))
+		res, err := gwc.Solve(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+
+		checkFile(ctx, t, filepath.Join(sourceName, "file.txt"), res, []byte("test contents\n"))
+		return gwclient.NewResult(), nil
+	})
+}
+
+func testSourceMountPathHandledByFilter(t *testing.T) {
```

Comment:
I realize this test isn't ideal since it relies on the network and a particular git repo, but this test is temporary. This test case could probably be handled by a test that checks protobuf directly like in #110, or an inline source once those are introduced

Reply:
Yeah, let's get #110 in and do these thests there, I think.

Reply:
Currently working on adding these in with the new source tests